galaxykit/collections.py

```diff
@@ -0,0 +1,112 @@
+"""
+Functions for managing collections. Currently only handles uploading test collections.
+"""
+
+import uuid
+import os
+import json
+from time import sleep
+from urllib.parse import urljoin
+from orionutils.generator import build_collection
+from .client import GalaxyClientError
+
+
+def upload_test_collection(client, namespace=None, collection_name=None):
+    """
+    Uploads a test collection generated with orionutils
+    """
+    config = {"namespace": namespace or client.username}
+    if collection_name is not None:
+        config["name"] = collection_name
+    artifact = build_collection(
+        "skeleton",
+        config=config,
+    )
+    upload_resp_url = upload_artifact(config, client, artifact)["task"]
+
+    ready = False
+    state = ""
+    while not ready:
+        sleep(1)
+        task_resp = client.get(upload_resp_url)
+        state = task_resp["state"]
+        ready = state in ["completed", "failed"]
+    if state == "failed":
+        raise GalaxyClientError(json.dumps(task_resp))
+    return task_resp
+
+
+def upload_artifact(
+    config,
+    client,
+    artifact,
+    hash=True,
+    no_filename=False,
+    no_file=False,
+    use_distribution=True,
+):
+    """
+    Publishes a collection to a Galaxy server and returns the import task URI.
+
+    :param collection_path: The path to the collection tarball to publish.
```

Comment:
The `param(s)` here are not accurate. Do not reflect the actual function parameters.

Reply:
Also, I know this code wasn't the best documented originally, but as long as we're bringing it over let's give the code a good commenting pass.

Reply:
:+1: will do